test-d/types/applications/forms/itemSheet.test-d.ts

```diff
@@ -0,0 +1,14 @@
+import '../../../../index';
+import { expectType } from 'tsd';
+
+type TestItemData = Item.Data<{ foo: string }>;
+class TestItem extends Item<TestItemData> {}
+const testItem = new TestItem();
+
+class TestItemSheet extends ItemSheet<TestItem> {}
+const testItemSheet = new TestItemSheet(testItem);
+
+const sheetData = await testItemSheet.getData();
+
+expectType<string>(sheetData.data.foo);
+expectType<string | undefined>(testItemSheet.actor?.data.items[0].data.foo);
```

Comment:
Does this even work? Because coming from Ruby and from some short testing, it you use conditional access (or optional chaining in JS), you normally have to make everything after the first one with conditional access as well, otherwise you will still get an error if the property is not defined.

Reply:
It works correctly: https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgLhgBTgJygSzgGwDwBKApqGgCZ7RqZgDmANDAN4BGSTw7ZSVNtAvoIB8QmAF5m-ANwAoGaEiwoRaOPgA6AGYgQAfnUt1wdWSlA